core/txbuilder/finalize.go

```diff
@@ -45,6 +45,9 @@ func FinalizeTx(ctx context.Context, c *protocol.Chain, s Submitter, tx *legacy.
 	if err != nil {
 		return errors.Wrap(err, "tx rejected")
 	}
+	if tx.Tx.MaxTimeMs < c.TimestampMS() {
+		return errors.Wrap(ErrRejected, "tx expired")
+	}
```

Comment:
I think this check already happens up in the submit handler: https://github.com/chain/chain/blob/main/core/transact.go#L273-L275